<?xml version="1.0" ?>
<robot name="trial2_robot" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!--Constants for robot dimensions-->
  <xacro:property name="PI" value="3.1415926535897931"/>
  <xacro:property name="density" value="100.0"/>
  <xacro:property name="axel_offset" value="0.05"/>
  <xacro:include filename="$(find robot_generator)/urdf/trial2_root/trial2_robot.gazebo"/>
  <xacro:include filename="$(find robot_generator)/urdf/materials.xacro"/>
  <!--World link for fixing robot to Gazebo frame-->
  <link name="world"/>
  <joint name="joint0" type="fixed">
    <parent link="world"/>
    <child link="link1"/>
  </joint>
  <!--Start of generated robot-->
  <link name="link1">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 1.0"/>
      <geometry>
        <box size="0.1 0.1 2"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 1.0"/>
      <geometry>
        <box size="0.1 0.1 2"/>
      </geometry>
      <material name="orange"/>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 1.0"/>
      <mass value="2.0"/>
      <inertia ixx="${2.0 / 12.0 * (0.1*0.1 + 2*2)}" ixy="0.0" ixz="0.0" iyy="${2.0 / 12.0 * (2*2 + 0.1*0.1)}" iyz="0.0" izz="${2.0 / 12.0 * (0.1*0.1 + 0.1*0.1)}"/>
    </inertial>
  </link>
  <link name="link2">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 ${0.5 - axel_offset}"/>
      <geometry>
        <box size="0.1 0.1 1.0"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 ${0.5 - axel_offset}"/>
      <geometry>
        <box size="0.1 0.1 1.0"/>
      </geometry>
      <material name="black"/>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 ${0.5 - axel_offset}"/>
      <mass value="1.0"/>
      <inertia ixx="${1.0 / 12.0 * (0.1*0.1 + 1.0*1.0)}" ixy="0.0" ixz="0.0" iyy="${1.0 / 12.0 * (1.0*1.0 + 0.1*0.1)}" iyz="0.0" izz="${1.0 / 12.0 * (0.1*0.1 + 0.1*0.1)}"/>
    </inertial>
  </link>
  <joint name="joint1" type="continuous">
    <parent link="link1"/>
    <child link="link2"/>
    <origin rpy="0 0 0" xyz="0 0.1 ${2 - axel_offset}"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0.7"/>
  </joint>
  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="link3">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 ${0.5 - axel_offset}"/>
      <geometry>
        <box size="0.1 0.1 1.0"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 ${0.5 - axel_offset}"/>
      <geometry>
        <box size="0.1 0.1 1.0"/>
      </geometry>
      <material name="orange"/>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 ${0.5 - axel_offset}"/>
      <mass value="1.0"/>
      <inertia ixx="${1.0 / 12.0 * (0.1*0.1 + 1.0*1.0)}" ixy="0.0" ixz="0.0" iyy="${1.0 / 12.0 * (1.0*1.0 + 0.1*0.1)}" iyz="0.0" izz="${1.0 / 12.0 * (0.1*0.1 + 0.1*0.1)}"/>
    </inertial>
  </link>
  <joint name="joint2" type="continuous">
    <parent link="link2"/>
    <child link="link3"/>
    <origin rpy="0 0 0" xyz="0 0.1 ${1.0 - axel_offset}"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0.7"/>
  </joint>
  <transmission name="tran2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
</robot>
